components/webcompat/webcompat_exceptions_browsertest.cc

```diff
@@ -0,0 +1,180 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+#include <string_view>
+
+#include "base/feature_list.h"
+#include "base/path_service.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/browser/brave_browser_process.h"
+#include "brave/components/brave_shields/content/browser/brave_shields_util.h"
+#include "brave/components/brave_shields/core/common/brave_shield_constants.h"
+#include "brave/components/brave_shields/core/common/features.h"
+#include "brave/components/constants/brave_paths.h"
+#include "brave/components/webcompat/features.h"
+#include "brave/components/webcompat/webcompat_exceptions_service.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/test/base/chrome_test_utils.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/content_settings/core/common/content_settings.h"
+#include "components/content_settings/core/common/content_settings_types.h"
+#include "components/google/core/common/google_switches.h"
+#include "components/network_session_configurator/common/network_switches.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/test_utils.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/default_handlers.h"
+#include "url/origin.h"
+
+#if BUILDFLAG(IS_ANDROID)
+#include "chrome/test/base/android/android_browser_test.h"
+#else
+#include "chrome/browser/ui/browser_navigator_params.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#endif
+
+using brave_shields::ControlType;
+using content_settings::mojom::ContentSettingsType;
+using enum ContentSettingsType;
+
+struct TestCase {
```

Comment:
pls put these and below into an anonymous namespace

Reply:
done

Reply:
done